facebook/delphiFacebook/R/contingency_tables.R

```diff
@@ -0,0 +1,1068 @@
+#'   calculations necessary.
+#'
+#' @return none
+#' 
+#' @importFrom parallel detectCores
+#' 
+#' @export
+run_contingency_tables <- function(params, aggregations)
+{
+  params <- update_params(params)
+  
+  cw_list <- produce_crosswalk_list(params$static_dir)
+  archive <- load_archive(params)
+  msg_df("archive data loaded", archive$input_data)
+  
+  input_data <- load_responses_all(params)
+  input_data <- filter_responses(input_data, params)
+  msg_df("response input data", input_data)
+  
+  input_data <- merge_responses(input_data, archive)
+  data_agg <- create_data_for_aggregatation(input_data)
+  
+  data_agg <- filter_data_for_aggregatation(data_agg, params, lead_days = 12)
+  data_agg <- join_weights(data_agg, params, weights = "step1")
+  msg_df("response data to aggregate", data_agg)
+  
+  ## Set default number of cores for mclapply to the total available number,
+  ## because we are greedy and this will typically run on a server.
+  if (params$parallel) {
+    cores <- detectCores()
+    
+    if (is.na(cores)) {
+      warning("Could not detect the number of CPU cores; parallel mode disabled")
+      params$parallel <- FALSE
+    } else {
+      options(mc.cores = cores)
+    }
+  }
+  
+  if ( "cids" %in% params$output )
+  {
+    write_cid(data_agg, "part_a", params)
+  }
+  if ( "archive" %in% params$output )
+  {
+    update_archive(input_data, archive, params)
+  }
+  
+  data_agg <- make_human_readable(data_agg)
+  aggregations <- get_aggs(params, aggregations)
+  
+  browser()
+  
+  if (nrow(aggregations) > 0) {
+    aggregate_aggs(data_agg, aggregations, cw_list, params)
+  }
+  
+}
+
+
+#' Rename question codes to informative descriptions
+#'
+#' Column names beginning with "b_" are binary (T/F/NA); with "t_" are user-
+#' entered text; with "n_" are user-entered numeric; with "mc_" are multiple
+#' choice (where only a single response can be selected); and with "ms_" are
+#' so-called multi-select, where multiple responses can be selected.
+#' 
+#' Only binary columns are mapped from response codes to real values. Multiple
+#' choice and multi-select questions use the original numeric response codes.
+#'
+#' @param input_data    Data frame of individual response data
+#' 
+#' @return Data frame with descriptive column names
+#' 
+#' @importFrom dplyr rename
+#' 
+#' @export
+make_human_readable <- function(input_data) {
+  # Named list of question numbers and str replacement names
+  map_old_new_names <- c(
+    ## free response
+    # Either number ("n"; can be averaged although may need processing) or text ("t")
+    "n_hh_num_sick" = "hh_number_sick", # A2
+    "n_hh_num_children" = "A5_1",
+    "n_hh_num_adults" = "A5_2",
+    "n_hh_num_seniors" = "A5_3",
+    # "t_zipcode" = "A3", # zip5
+    "n_cmnty_num_sick" = "A4",
+    "t_symptoms_other" = "B2_14_TEXT",
+    "t_unusual_symptoms_other" = "B2c_14_TEXT",
+    "t_gender_other" = "D1_4_TEXT",
```

Comment:
We likely won't produce any aggregates that use text responses, since users may enter identifiable information in these fields

Reply:
I provided descriptive names for all existing columns for completeness, regardless of intent to use columns in aggregations. Would we prefer to drop or just not rename those we don't expect to be used?